pkg/engine/jmespath/functions_test.go

```diff
@@ -391,63 +391,395 @@ func Test_LabelMatch(t *testing.T) {
 }
 
 func Test_Add(t *testing.T) {
-	jp, err := New("add(`12`, `13`)")
-	assert.NilError(t, err)
+	testCases := []struct {
+		test           string
+		expectedResult interface{}
+		err            bool
+		retFloat       bool
+	}{
+		{
+			test: "add('12', '13s')",
+			err:  true,
+		},
+		{
+			test: "add('12Ki', '13s')",
+			err:  true,
+		},
+		{
+			test: "add('12s', '13')",
+			err:  true,
+		},
+		{
+			test: "add('12s', '13Ki')",
+			err:  true,
+		},
+		{
+			test:           "add(`12`, `13`)",
+			expectedResult: 25.0,
+			retFloat:       true,
+		},
+		{
+			test:           "add(`12`, '13s')",
+			expectedResult: `25s`,
+		},
```

Comment:
Do we really get the following case after parsing the YAML / JSON file? Or is it just for testing purposes?
```
		{
			test: "add('12', '13s')",
			err:  true,
		},
```

Reply:
This is just a fail-safe, in case the user tried to work operate with one operand as a duration and the other as a resource quantity.

Reply:
So `12` here is the quantity?

Reply:
Yes, for resource quantities we don't write Byte explicitly, we write `128Ki`/`128Gi` which is a number so a number passed as a string is treated as a resource quantity.

Reply:
Got it! I thought it represents a duration.